var/spack/repos/builtin/packages/conduit/package.py

```diff
@@ -0,0 +1,312 @@
+from spack import *
+
+import socket
+import os
+
+import llnl.util.tty as tty
+
+
+def cmake_cache_entry(name, value):
+    """
+    Helper that creates CMake cache entry strings used in
+    'host-config' files.
+    """
+    return 'set("{0}" "{1}" CACHE PATH "")\n\n'.format(name, value)
+
+
+class Conduit(Package):
+    """Conduit is an open source project from Lawrence Livermore National
+    Laboratory that provides an intuitive model for describing hierarchical
+    scientific data in C++, C, Fortran, and Python. It is used for data
+    coupling between packages in-core, serialization, and I/O tasks."""
+
+    homepage = "http://software.llnl.gov/conduit"
+    url = "https://github.com/LLNL/conduit/archive/v0.2.0.tar.gz"
+
+    version('0.2.0', 'd595573dedf55514c11d7391092fd760')
+
+    version('master', git='https://github.com/LLNL/conduit.git')
+
+    ###########################################################################
+    # package variants
+    ###########################################################################
+
+    variant("shared", default=True, description="Build Conduit as shared libs")
+
+    variant("cmake", default=True,
+            description="Build CMake (if off, attempt to use cmake from PATH)")
+
+    # variants for python support
+    variant("python", default=True, description="Build Conduit Python support")
+
+    # variants for comm and i/o
+    variant("mpi", default=True, description="Build Conduit MPI Support")
+    variant("hdf5", default=True, description="Build Conduit HDF5 support")
+    variant("silo", default=True, description="Build Conduit Silo support")
+
+    # variants for dev-tools (docs, etc)
+    variant("doc", default=False, description="Build Conduit's documentation")
+
+    ###########################################################################
+    # package dependencies
+    ###########################################################################
+
+    #######################
+    # CMake
+    #######################
+    # cmake 3.3.1 is the version we tested
+    depends_on("cmake@3.3.1", when="+cmake")
+
+    #######################
+    # Python
+    #######################
+    extends("python", when="+python")
+    # TODO: blas and lapack are disabled due to build
+    # issues Cyrus experienced on OSX 10.11.6
+    depends_on("py-numpy~blas~lapack", when="+python")
+
+    #######################
+    # I/O Packages
+    #######################
+    # TODO: cxx variant is disabled due to build issue Cyrus
+    # experienced on BGQ. When on, the static build tries
+    # to link agains shared libs.
+    #
+    # we are not using hdf5's mpi or fortran features.
+    depends_on("hdf5~cxx~mpi~fortran", when="+shared")
+    depends_on("hdf5~shared~cxx~mpi~fortran", when="~shared")
+
+    # we are not using silo's fortran features
+    depends_on("silo~fortran", when="+shared")
+    depends_on("silo~shared~fortran", when="~shared")
```

Comment:
@cyrush @white238 
I get that you are not using these features, but wouldn’t conduit work the same with this features?
If that is the case, I think for the sake of having as little constraint as possible of the spec we should no specify the mpi variant in hdf5.
This would allow to concretize and build a working version of conduit together with other products needing the mpi variant to be "on".
Does it makes sense to you?

Reply:
@adrienbernede, yes, I am ok with relaxing.

Note: this type change to conduit is  already part of the ceed PR:

https://github.com/spack/spack/pull/15500/files#r395829308

Maybe verify those changes look good?

Reply:
Thanks! Looks great and already merged in Spack, wonderful!